crates/lib/src/rules/aliasing/al05.rs

```diff
@@ -159,10 +176,53 @@ impl RuleAL05 {
                     .extend(select_info.table_aliases);
 
                 for r in select_info.reference_buffer {
-                    for tr in
-                        r.extract_possible_references(ObjectReferenceLevel::Table, dialect.name)
-                    {
-                        Self::resolve_and_mark_reference(query.clone(), tr.part);
+                    // Standard table reference extraction
+                    let table_refs = r.extract_possible_references(ObjectReferenceLevel::Table, dialect.name);
+                    for tr in &table_refs {
+                        Self::resolve_and_mark_reference(query.clone(), tr.part.clone());
+                    }
+                    
+                    // For T-SQL, be more aggressive about extracting references from qualified names
+                    if dialect.name == DialectKind::Tsql {
```

Comment:
I would need some help on this one, its now working as intended, but I was not sure if this could effect others.
Therefor I limited it to tsql only, what do you think?
Can I remove this to allow it on all?

Tests:
https://github.com/quarylabs/sqruff/blob/65f16364a4ef9fbdfd0ec4088bef51c215c99c2f/crates/lib/test/fixtures/rules/std_rule_cases/AL05.yml#L449-L494